main.go

```diff
@@ -15,6 +15,138 @@
 
 package main
 
+import (
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/exec"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/spf13/cast"
+
+	"github.com/bank-vaults/secret-init/logger"
+	"github.com/bank-vaults/secret-init/providers"
+	"github.com/bank-vaults/secret-init/providers/vault"
+)
+
+var providersMap = map[string]providers.Provider{
+	// "aws": aws.NewAWSProvider(),
+	// "gcp": gcp.NewGCPProvider(),
+	"vault": vault.NewVaultProvider(),
+}
+
 func main() {
-	println("hello world")
+	logger := logger.SetupSlog()
+
+	var providerName string
+	if len(os.Args) >= 3 && os.Args[1] == "-p" {
```

Comment:
This is a very...simplistic approach to parsing flags.

Probably pflag is the way to go: https://github.com/spf13/pflag/